benchmarks/bench_importasdm.py

```diff
@@ -0,0 +1,243 @@
+
+import os
+import sys
+import shutil
+
+from casatools import ctsys
+from casatasks import importasdm, casalog
+
+# ASV iteration control (https://asv.readthedocs.io/en/stable/benchmarks.html#benchmark-attributes)
+number = 1            # i.e., always run the setup and teardown methods
+repeat = (1, 2, 30.0) # between 1 and 2 iterations per round w/ soft cutoff (start no new repeats) past 1m
+rounds = 3            # amount of instances a "repeat block" is run to collect samples
+min_run_count = 5     # enforce the min_repeat * rounds setting is met
+timeout = 3600        # conservative 1hr hard cap for duration of a single test execution
+
+datapath = ctsys.resolve('unittest/importasdm/')
+
+# benchmarks grouped by ASDM used
+
+class M51():
+    # the test code says this is an M51 ASDM but the source name and direction are for 1924-202, which is a quasar
+    asdm_name = 'uid___X5f_X18951_X1'
+    ms_name = asdm_name + '.ms'
+
+    def setup(self):
+        shutil.copytree(os.path.join(datapath,self.asdm_name), self.asdm_name)
+
+    def teardown(self):
+        shutil.rmtree(self.asdm_name)
+        shutil.rmtree(self.ms_name)
+        shutil.rmtree(self.ms_name+'.flagversions')
+
+    def time_all_defaults(self):
+        '''test_import2 --- importasdm with default arguments'''
+        importasdm(self.asdm_name)
+
+    def time_lazy_pc(self):
+        '''test1_lazy1 --- lazy mode, with pointing_correction=True'''
+        importasdm(asdm=self.asdm_name, vis=self.ms_name, lazy=True, with_pointing_correction=True)
+
+    def time_pc(self):
+        '''test1_lazy1 --- standard fill (lazy=False), with pointing_correction=True'''
+        importasdm(asdm=self.asdm_name, vis=self.ms_name, lazy=False, with_pointing_correction=True)
+
+class Evla():
+    # EVLA sdm
+    asdm_name = 'X_osro_013.55979.93803716435'
+    ms_name = asdm_name + '.ms'
+    cmdfile = ms_name.replace('.ms','_cmd.txt')
+
+    def setup(self):
+        shutil.copytree(os.path.join(datapath,self.asdm_name), self.asdm_name)
+
+    def teardown(self):
+        shutil.rmtree(self.asdm_name)
+        shutil.rmtree(self.ms_name)
+        if (os.path.exists(self.ms_name+'.flagversions')):
+            shutil.rmtree(self.ms_name+'.flagversions')
+        if (os.path.exists(self.cmdfile)):
+            os.remove(self.cmdfile)        
+    
+    def time_evlatest1(self):
+        '''test_evlatest1 - test of importing evla data, with_pointint_correction=True is normal for EVLA data'''
+
+        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='2',ocorr_mode='co',with_pointing_correction=True)
+
+    def time_evla_apply2(self):
+        '''test_evla_apply2 - test of importing evla data and applying the online flags'''
+
+        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='2',ocorr_mode='co', with_pointing_correction=True,
+                   process_flags=True, applyflags=True, savecmds=False, flagbackup=False)
+
+    def time_evla_apply3(self):
```

Comment:
suggest renaming to something such as time_scans or time_scan_selection. It is not applying the flags on the fly therefore the word "apply" should not appear.